src/web/src/app/space/add/space-key-validator.ts

```diff
@@ -13,9 +14,16 @@ export class SpaceKeyValidatorDirective implements AsyncValidator {
 	constructor(private space: SpaceService) {}
 
 	validate(control: AbstractControl): Promise<ValidationErrors | null> | Observable<ValidationErrors | null> {
-		return this.space.isExists(control.value)
-			.pipe(map(res => {
-				return res ? { 'keyExists': true} : null;
-			}));
+		return timer(500).pipe(switchMap(()=>{
+			return this.space.isExists(control.value)
+				.pipe(
+					map(res => {
+						if (res == null) {
+							return null;
+						}
+						return { 'keyExists': true };
+					})
```

Comment:
res - у нас уже должен быть boolean, давай его в значении и верём.
```suggestion
					map(res => ({ 'keyExists': res }))
```

Reply:
Готово.